src/main/java/com/accantosystems/stratoss/vnfmdriver/web/etsi/GrantController.java

```diff
@@ -39,11 +45,13 @@ public GrantController(GrantService grantService) {
     @ApiOperation(value = "Requests a grant for a particular VNF lifecycle operation.", code = 201)
     public ResponseEntity<Grant> requestGrant(@RequestBody GrantRequest grantRequest) throws GrantRejectedException, GrantProviderException {
         logger.info("Received grant request:\n{}", grantRequest);
-
+        UUID uuid = UUID.randomUUID();
+        LoggingUtils.logEnabledMDC(grantRequest.toString(), MessageType.REQUEST, MessageDirection.RECEIVED, uuid.toString(),MediaType.APPLICATION_JSON.toString(), "https",getRequestProtocolMetaData(GRANT_LOCATION) ,null);
```

Comment:
You might need to handle the null on grantRequest as well in the first parameter